shims/spark311cdh/src/main/scala/com/nvidia/spark/rapids/shims/spark311cdh/Spark311CDHShims.scala

```diff
@@ -41,4 +49,107 @@ class Spark311CDHShims extends Spark31XShims with Spark30Xuntil33XShims {
     new ParquetFilters(schema, pushDownDate, pushDownTimestamp, pushDownDecimal, pushDownStartWith,
       pushDownInFilterThreshold, caseSensitive)
   }
+
+  /**
+   * create Orc OutStream
+   * CDH version use the shaded Orc 1.5.10 code
+   * @return
+   */
+  override def newOrcOutStream(name: String, orcBufferSize: Int, codec: CompressionCodec,
```

Comment:
We have been moving away from using a runtime lookup of the shims (i.e.: methods in the shim interface as done here) to a compile-time lookup of the shims (i.e.: the shims v2 approach).  I'd prefer we use that here.  For example, we could create an OrcShim or OrcWrapper object class under com.nvidia.spark.rapids.shims.v2 that contains these ORC functions.  We would then create separate ones based on ORC version under the sql-plugin  directories and add those source paths to the appropriate profiles based on which ORC version the corresponding Spark version uses.  The common code would then simply use the OrcShim or OrcWrapper object directly rather than using the shim loader at runtime to lookup the shims.

Reply:
Yes, it is right, updated.